File: src/main/java/com/nndmove/app/service/dto/MovieGenresDTO.java
package com.nndmove.app.service.dto;

import java.io.Serializable;
import java.util.Objects;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * A DTO for the {@link com.nndmove.app.domain.MovieGenres} entity.
 */
@SuppressWarnings("common-java:DuplicatedBlocks")
@Getter
@Setter
@NoArgsConstructor
public class MovieGenresDTO implements Serializable {

    private Long id;

    private MovieDTO movie;

    private GenresDTO genres;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof MovieGenresDTO)) {
            return false;
        }

        MovieGenresDTO movieGenresDTO = (MovieGenresDTO) o;
        if (this.id == null) {
            return false;
        }
        return Objects.equals(this.id, movieGenresDTO.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.id);
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "MovieGenresDTO{" +
            "id=" + getId() +
            ", movie=" + getMovie() +
            ", genres=" + getGenres() +
            "}";
    }
}

File: src/main/java/com/nndmove/app/service/dto/MovieGenresDTO.kt
package com.nndmove.app.service.dto

import java.io.Serializable
import java.util.*

/**
 * A DTO for the [com.nndmove.app.domain.MovieGenres] entity.
 */
class MovieGenresDTO : Serializable {
    @JvmField
    var id: Long? = null

    @JvmField
    var movie: MovieDTO? = null

    @JvmField
    var genres: GenresDTO? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o !is MovieGenresDTO) {
            return false
        }

        if (this.id == null) {
            return false
        }
        return this.id == o.id
    }

    override fun hashCode(): Int {
        return Objects.hash(this.id)
    }

    // prettier-ignore
    override fun toString(): String {
        return "MovieGenresDTO{" +
                "id=" + id +
                ", movie=" + movie +
                ", genres=" + genres +
                "}"
    }
}
